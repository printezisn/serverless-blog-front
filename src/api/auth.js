import { CognitoAuth } from "amazon-cognito-auth-js"
import { Events } from "../utils/constants"
import { eventBus } from "../utils/eventBus"

class Auth {
    constructor() {
        const rootUrl = process.env.GATSBY_EDNABLOG_SITE_ROOT_URL
        const authData = {
            ClientId: "7icdk42lkcj7fevgp2lgss0aio",
            AppWebDomain: "ednablog.auth.ca-central-1.amazoncognito.com",
            TokenScopesArray: ["email", "openid"],
            RedirectUriSignIn: rootUrl,
            RedirectUriSignOut: rootUrl,
            UserPoolId: "ca-central-1_zPiF08WsY",
            AdvancedSecurityDataCollectionFlag: false,
        }

        const self = this
        const auth = new CognitoAuth(authData)
        auth.userhandler = {
            onSuccess: result => {
                self.accessToken = result.getAccessToken().getJwtToken()
                eventBus.emit(Events.AUTH_CHANGE)
            },
            onFailure: () => {},
        }

        this.auth = auth
        this.accessToken = this.auth
            .getSignInUserSession()
            .getAccessToken()
            .getJwtToken()
    }

    logIn() {
        this.auth.getSession()
    }

    logOut() {
        this.auth.signOut()
    }

    init(url) {
        this.auth.parseCognitoWebResponse(url)
    }

    getAccessToken() {
        return this.accessToken
    }

    isLoggedIn() {
        return this.accessToken !== ""
    }
}

export const auth = new Auth()
