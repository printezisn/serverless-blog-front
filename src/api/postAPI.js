class PostAPI {
    constructor() {
        this.root = process.env.GATSBY_EDNABLOG_API_ROOT_URL
    }

    async createPost(post) {
        const response = await fetch(this.root, {
            method: "PUT",
            mode: "cors",
            body: JSON.stringify(post),
        })

        return response.json()
    }

    async updatePost(post) {
        const response = await fetch(this.root, {
            method: "POST",
            mode: "cors",
            body: JSON.stringify(post),
        })

        return response.json()
    }

    async getPost(id) {
        const response = await fetch(this.root + encodeURIComponent(id), {
            method: "GET",
            mode: "cors",
        })

        return response.json()
    }

    async deletePost(id) {
        const response = await fetch(this.root + encodeURIComponent(id), {
            method: "DELETE",
            mode: "cors",
        })

        return response.json()
    }
}

export const postAPI = new PostAPI()
