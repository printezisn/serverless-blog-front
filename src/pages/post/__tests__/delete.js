import React from "react"
import renderer from "react-test-renderer"

jest.mock("@reach/router", () => {
    const location = {
        search: "?id=test-id",
    }

    return {
        Location: props => <p>{props.children({ location })}</p>,
    }
})
jest.mock("../../../components/layout", () => "Layout")
jest.mock("../../../components/post", () => {
    return {
        PostLoader: "PostLoader",
    }
})
jest.mock("../../../components/markdown", () => "Markdown")
jest.mock("../../../components/messageBox", () => "MessageBox")
jest.mock("../../../components/loadingButton", () => "LoadingButton")
jest.mock("../../../api/postAPI", () => {
    return {
        postAPI: {
            deletePost: jest.fn(),
        },
    }
})

import { postAPI } from "../../../api/postAPI"
import DeletePage from "../delete"

describe("Delete page", () => {
    it("renders correctly", () => {
        const deletePage = renderer.create(<DeletePage />)

        expect(deletePage.toJSON()).toMatchSnapshot()
    })

    it("renders correctly when the post is loaded", async () => {
        const post = {
            id: "test-id",
            title: "test-title",
            description: "test-description",
            body: "test-body",
        }
        const deletePage = renderer.create(<DeletePage />)
        const postComponent = await deletePage.root
            .findByType("PostLoader")
            .props.onRender(post)

        expect(postComponent).toMatchSnapshot()
    })

    it("renders correctly when the post is not found", async () => {
        const post = {
            id: "test-id",
            title: "test-title",
            description: "test-description",
            body: "test-body",
        }
        const deletePage = renderer.create(<DeletePage />)
        let postComponent = renderer.create(
            await deletePage.root.findByType("PostLoader").props.onRender(post)
        )

        postAPI.deletePost.mockReturnValue({ StatusCode: 404 })

        await postComponent.root.findByType("LoadingButton").props.onClick()

        postComponent = await deletePage.root
            .findByType("PostLoader")
            .props.onRender(post)

        expect(postAPI.deletePost).toHaveBeenCalledWith(post.id)
        expect(postComponent).toMatchSnapshot()
    })

    it("renders correctly when an unexpected error occurs", async () => {
        const post = {
            id: "test-id",
            title: "test-title",
            description: "test-description",
            body: "test-body",
        }
        const deletePage = renderer.create(<DeletePage />)
        let postComponent = renderer.create(
            await deletePage.root.findByType("PostLoader").props.onRender(post)
        )

        postAPI.deletePost.mockReturnValue({ StatusCode: 400 })

        await postComponent.root.findByType("LoadingButton").props.onClick()

        postComponent = await deletePage.root
            .findByType("PostLoader")
            .props.onRender(post)

        expect(postAPI.deletePost).toHaveBeenCalledWith(post.id)
        expect(postComponent).toMatchSnapshot()
    })

    it("renders correctly when the post is deleted successfully", async () => {
        const post = {
            id: "test-id",
            title: "test-title",
            description: "test-description",
            body: "test-body",
        }
        const deletePage = renderer.create(<DeletePage />)
        let postComponent = renderer.create(
            await deletePage.root.findByType("PostLoader").props.onRender(post)
        )

        postAPI.deletePost.mockReturnValue({ StatusCode: 200 })

        await postComponent.root.findByType("LoadingButton").props.onClick()

        postComponent = await deletePage.root
            .findByType("PostLoader")
            .props.onRender(post)

        expect(postAPI.deletePost).toHaveBeenCalledWith(post.id)
        expect(postComponent).toMatchSnapshot()
    })
})
