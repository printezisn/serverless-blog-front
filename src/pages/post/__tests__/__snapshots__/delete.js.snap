// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete page renders correctly 1`] = `
<Router>
  <Layout
    compact={true}
  >
    <PostLoader
      onRender={[Function]}
    />
  </Layout>
</Router>
`;

exports[`Delete page renders correctly when an unexpected error occurs 1`] = `
<article>
  <header>
    <h2
      className="title is-3"
    >
      test-title
    </h2>
  </header>
  <MessageBox
    isForErrors={true}
    messages={
      Array [
        "An error occurred. Please try again later.",
      ]
    }
  />
  <MessageBox
    isForErrors={false}
    messages={Array []}
  />
  <section
    className="markdown"
  >
    <Markdown
      text="test-body"
    />
  </section>
  <br />
  <footer>
    <LoadingButton
      className="button is-danger"
      isLoading={false}
      onClick={[Function]}
      type="button"
    >
      <i
        className="fas fa-trash-alt"
      />
        Delete
    </LoadingButton>
  </footer>
</article>
`;

exports[`Delete page renders correctly when the post is deleted successfully 1`] = `
<article>
  <header>
    <h2
      className="title is-3"
    >
      test-title
    </h2>
  </header>
  <MessageBox
    isForErrors={true}
    messages={Array []}
  />
  <MessageBox
    isForErrors={false}
    messages={
      Array [
        "The post was deleted successfully!",
      ]
    }
  />
  <section
    className="markdown"
  >
    <Markdown
      text="test-body"
    />
  </section>
  <br />
  <footer>
    <LoadingButton
      className="button is-danger"
      isLoading={false}
      onClick={[Function]}
      type="button"
    >
      <i
        className="fas fa-trash-alt"
      />
        Delete
    </LoadingButton>
  </footer>
</article>
`;

exports[`Delete page renders correctly when the post is loaded 1`] = `
<article>
  <header>
    <h2
      className="title is-3"
    >
      test-title
    </h2>
  </header>
  <MessageBox
    isForErrors={true}
    messages={Array []}
  />
  <MessageBox
    isForErrors={false}
    messages={Array []}
  />
  <section
    className="markdown"
  >
    <Markdown
      text="test-body"
    />
  </section>
  <br />
  <footer>
    <LoadingButton
      className="button is-danger"
      isLoading={false}
      onClick={[Function]}
      type="button"
    >
      <i
        className="fas fa-trash-alt"
      />
        Delete
    </LoadingButton>
  </footer>
</article>
`;

exports[`Delete page renders correctly when the post is not found 1`] = `
<article>
  <header>
    <h2
      className="title is-3"
    >
      test-title
    </h2>
  </header>
  <MessageBox
    isForErrors={true}
    messages={
      Array [
        "The blog post you requested was not found.",
      ]
    }
  />
  <MessageBox
    isForErrors={false}
    messages={Array []}
  />
  <section
    className="markdown"
  >
    <Markdown
      text="test-body"
    />
  </section>
  <br />
  <footer>
    <LoadingButton
      className="button is-danger"
      isLoading={false}
      onClick={[Function]}
      type="button"
    >
      <i
        className="fas fa-trash-alt"
      />
        Delete
    </LoadingButton>
  </footer>
</article>
`;
